packages/core/buckets/offer-store.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  S3Client,
+  ServiceInputTypes,
+  PutObjectCommand,
+} from '@aws-sdk/client-s3'
+import pRetry from 'p-retry'
+
+import * as AggregateAPI from '@web3-storage/aggregate-api'
+import { ArrangedOfferStore } from '../tables/arranged-offer-store'
+
+export function createOfferStore(region: string, bucketName: string, arrangedOfferStore: ArrangedOfferStore, options?: ServiceInputTypes) {
+  const s3 = new S3Client({
+    region,
+    ...(options || {}),
+  })
+  return useOfferStore(s3, bucketName, arrangedOfferStore)
+}
+
+/**
+ * Offer Store bucket will persist offers keyed as `YYYY-MM-DD HH:MM:00 `${commitmentProof}``.
+ * The time format follows same pattern as used by AWS Athena + AWS S3
+ */
+
+export function useOfferStore(s3client: S3Client, bucketName: string, arrangedOfferStore: ArrangedOfferStore) {
+  return {
+    queue: async (aggregateOffer: AggregateAPI.OfferToQueue) => {
+      const { commitmentProof, offers } = aggregateOffer
+      const putCmd = new PutObjectCommand({
+        Bucket: bucketName,
+        ContentType: 'application/json',
+        Key: `${getNextUtcDateName()} ${commitmentProof.toString()}`,
+        Body: JSON.stringify({
+          commitmentProof: commitmentProof.toString(),
+          offers
+        })
+      })
+
+      await pRetry(() => s3client.send(putCmd))
```

Comment:
this is for historical reasons where sometimes write fail in these buckets. So we usually wrap with retires across the stack, so that spontaneous issues do not break calls